docs/source/guides/data.rst

```diff
@@ -357,3 +359,21 @@ option when using sequential data.
 
     # Set limit_val_batches as an int
     trainer = Trainer(limit_val_batches=100)
+
+
+*****************************************
+Accessing DataLoaders in LightningModules
+*****************************************
+
+In the case that you require access to the DataLoader or Dataset objects, DataLoaders for each step can be accessed using the `Trainer` object:
+
+.. code-block:: python
+
+    def test_step(batch, batch_idx, dataloader_idx):
+        test_dl = self.trainer.test_dataloaders[dataloader_idx]
```

Comment:
I think we should add example when using `CombinedLoader` and one for train dataloader. Train dataloader is always a combined dataloader
When someone use `CombinedLoader`, the dataloaders are accessible using `self.test_dataloader.loaders`

Reply:
I'm not familiar with the internals of `CombinedLoader` -- if you don't mine providing an example I'd be happy to add it here

Reply:
Here is an example:

```py
def training_step(self, batch, batch_idx):
        train_dl = self.trainer.train_dataloadesr.loaders[0]
```

Reply:
same we can extend for eval dataloaders having dict return from eval_dataloader

```py
from pytorch_lightning.trainer.supporters import CombinedLoader

def val_dataloader(self):
    val_dl1 = ...
    val_dl2 = ...
    return CombinedLoader({'dl1: val_dl1, 'dl2': val_dl2})

def validation_step(self, batch, batch_idx):
    val_dl1 = self.val_dataloaders[0]['val_dl1'] 
```

Reply:
It should be self.val_dataloaders.loaders['val_dl_1']